chore(alias): upgrade TypeScript

benmccann opened the pull request:
<!--
  ⚡️ katchow! We ❤️ Pull Requests!

  If you remove or skip this template, you'll make the 🐼 sad and the mighty god
  of Github will appear and pile-drive the close button from a great height
  while making animal noises.

  Pull Request Requirements:
  * Please include tests to illustrate the problem this PR resolves.
  * Please lint your changes by running `npm run lint` before creating a PR.
  * Please update the documentation in `/docs` where necessary

  Please place an x (no spaces - [x]) in all [ ] that apply.
-->

<!-- the plugin(s) this PR is for -->

## Rollup Plugin Name: `alias`

This PR contains:

- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [X] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [X] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [X] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

### Description

<!--
  Please be thorough and clearly explain the problem being solved.
  * If this PR adds a feature, look for previous discussion on the feature by searching the issues first.
  * Is this PR related to an issue?
-->

benmccann commented:
`lint` fails on `master` for me. It seems unrelated to this change

shellscape commented:
Let's focus on one plugin at a time here, especially since there are linting errors abound. #717 can be closed (just did that), and the work merged into this branch. Let's solve the TS linting error here, since it's likely to show up again elsewhere.

All of the other TS upgrade PRs are going to be on hold until we move this one forward. After this one, we'll tackle the next, and so on.

benmccann commented:
Ok, I've updated this PR to address the lint issues as well. It was trying to use a JavaScript rule on TypeScript code. I fixed that here, but also sent the same change to the base config: https://github.com/rollup/eslint-config-rollup/pull/3

shellscape commented on .eslintrc.js:
```diff
@@ -32,6 +33,7 @@ module.exports = {
     ],
     'import/no-namespace': 'off',
     'import/no-named-export': 'off',
+    'no-redeclare': 'off',
```
that's not the right fix. see https://github.com/rollup/eslint-config-rollup/pull/3/files#r538998040

benmccann replied:
this actually does seem like the right fix to me per the conversation on the thread linked in the above comment

benmccann commented:
@shellscape if you have a chance to take a look at these PRs I think it'd be most helpful to prioritize this one and https://github.com/rollup/plugins/pull/715 in order to fix the lint errors that are happening on `master`
